app/models/mixins/process_tasks_mixin.rb

```diff
@@ -22,12 +22,12 @@ def invoke_tasks_queue(options)
 
     # Performs tasks received from the UI via the queue
     def invoke_tasks(options)
-      local, remote = partition_ids_by_remote_region(options[:ids])
+      local, remote = partition_ids_by_remote_region(options[:ids].map(&:to_i))
       invoke_tasks_local(options.merge(:ids => local)) unless local.empty?
 
       # TODO: invoke_tasks_remote currently is only implemented by VmOrTemplate.
       # it can be refactored to be generalized like invoke_tasks_local
-      invoke_tasks_remote(options.merge(:ids => remote)) if remote.present? && respond_to?("invoke_tasks_remote")
+      invoke_tasks_remote_queue(options.merge(:ids => remote)) if remote.present? && respond_to?("invoke_tasks_remote_queue")
```

Comment:
@carbonin Not for now, but I wonder if it makes sense to wrap this process in a MiqTask object that the UI can display in the my tasks area... the silly named (thanks Me from 5+ years ago) `MiqTask.generic_action_with_callback` basically does this... returning the task id so the UI can choose to spin (waiting for the task to complete) or they can just show it in the tasks area...